lib/generators/rolemodel/webpacker/templates/generated/config/webpack/development.js

```diff
@@ -2,4 +2,20 @@ process.env.NODE_ENV = process.env.NODE_ENV || 'development'
 
 const environment = require('./environment')
 
+// This will cause source maps to be generated and a file:/// sourceMappingURL
+// will be appended to the bundled JS
+environment.config.devtool = false
+const { SourceMapDevToolPlugin } = require('webpack')
+environment.plugins.prepend(
+  'SourceMapDevToolPlugin',
+  new SourceMapDevToolPlugin({
+    append: "\n//# sourceMappingURL=[url]",
```

Comment:
I'm fairly certain that this will need to be

```javascript
   append: "\n//# sourceMappingURL=/maps/[url]",
```

since [url] will be substituted here with the name of the map file.

Reply:
It was intentional. So, if this comment contains only the file name, web-dev-tool will try to fetch the source map from the same directory as the original js bundle file. I don't want to expose `maps` folder to everyone. So, the middleware takes care of mapping URL to the `maps` folder.

Reply:
Nice! I didn't catch that behavior in the middleware class.

Reply:
One more thing...

The sourceMappingURL form that is in place here is the same as it would be with

```javascript
environment.config.devtool = 'source-map'
```

So, there is no need to use the `SourceMapDevToolPlugin` to achieve this outcome.

Also, one downside of using the middleware approach to guard the source maps is that this mechanism won't work for situations where we don't have a logged in user. This would be the case for the LightningCAD DPQ guest editor and may apply to other applications as well. For those situations, the source maps would have to be downloaded and manually applied.

Reply:
@tingraldi, we need to obtain a user ID to confirm that the current user can see the source map. I believe the only IP address could be used as a user ID in the 'guest' case. But the problem is most internet providers assign dynamic IP addresses. So, developers must update the 'allowed IP' list whenever new IP addresses are assigned.  It is possible to build a script for updating the 'allowed IP' list, but I don't think it is worth it (We do not have a lot of cases like the 'LightningCAD DPQ guest editor' one). So, I agree with downloading and applying manually to web-dev-tool in similar cases.

Reply:
Oh, I think that I should move WebPack configs to the production environment =)

Reply:
Correct!

Reply:
Resolved.